### DISCUSSION TITLE
📚 Add filter hook for limiting the list of fonts returned in the Font Family control

### DISCUSSION THREAD

**Discussion by rbezant-drtavel** [elementor/elementor#28306] [Feature Request]
### Prerequisites

- [X] I have searched for similar feature requests in both open and closed discussions and cannot find an existing request.
- [X] I have verified that the feature is still missing from the latest stable versions of Elementor or Elementor Pro.

### What problem is your feature request going to solve? Please describe..

Many websites have specific branding guidelines that don't need, and never will need, the massive list of fonts that come up in the Font control (especially when google fonts is enabled). Rather, they would only utilize one or two fonts at most for the duration of that website's existence until a rebrand comes along. When introducing staff to create content that might not be as meticulous or aware of the branding guidelines, the margin for error in font selection would be vastly reduced if we could filter out all but our desired fonts programmatically.

### Describe the solution you'd like.

A Wordpress filter placed in \Elementor\Fonts::get_fonts() after get_native_fonts() is called, that allows filtering all fonts that would be returned. This could ultimately replace the `elementor/fonts/additional_fonts` filter.

### Describe alternatives you've considered.

Elementor's Google Fonts feature could be disabled in the settings and then specific fonts added as custom fonts. It's a more manual solution on the end-user side rather than the developer side that requires downloading and installing fonts server-side (which might not be possible for some fonts due to licensing restrictions or if the use of a CDN is preferred). It also doesn't allow as fine-grained control of the fonts returned in the list, still allowing the system fonts and fonts from other sources to be selected.

### Additional context

The filter could be called `elementor/fonts/get_fonts` and might be placed in the code like so:

/includes/fonts.php
```php
	public static function get_fonts() {
		if ( null === self::$fonts ) {
			$additional_fonts = [];

			/**
			 * Additional fonts.
			 *
			 * Filters the fonts used by Elementor to add additional fonts.
			 *
			 * @since 1.9.4
			 *
			 * @param array $additional_fonts Additional Elementor fonts.
			 */
			$additional_fonts = apply_filters( 'elementor/fonts/additional_fonts', $additional_fonts );

			 // Filter placed here
			self::$fonts = apply_filters( 'elementor/fonts/get_fonts', array_replace( self::get_native_fonts(), $additional_fonts ) );
		}

		return self::$fonts;
	}
```

### Agreement

- [X] I agree that my request may be closed without action if it doesn't meet all the requirements.

**Comment by hitsgee** [comment]
I was just thinking about this the other day. Seems simple enough to add. Performance wise it should have some minor impact on editor load.
